### DISCUSSION TITLE
Adding additional persistent directories for derivatives

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [kairos-io/kairos#481] [Q&A]
Hey there! Is there a simple way to add a few persistent directories? I have a few applications that write config to etc at runtime and I have to persist that data. I don't really want to overwrite the whole /system/oem/11_persistency.yaml...would prefer to just add a .yaml higher in the stack to add that directory. Is that possible?

**Comment by mudler** [comment]
Hey! 

Good point indeed, currently you can just copy that file, and rename it with a higher number, that should guarantee it hits in order and would overwrite the defaults. That however, will overwrite Kairos defaults with yours (so changes in Kairos won't anymore take effect).

For the future, we have https://github.com/kairos-io/kairos/issues/210 in our backlog which is meant to cover exactly this scenario.

**Comment by fullmetal-fred** [comment]
I suppose one could always add a few sed commands in the derivative's dockerfile to add the additional directories to 11_persistency without fully overwriting the file.
________________________________

**Reply by mudler** [reply]
yup right, that would work as well - however that's less declarative , ideally we would like to make that more streamlined from the config file with https://github.com/kairos-io/kairos/issues/210
